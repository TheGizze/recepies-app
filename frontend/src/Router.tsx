import React from 'react';
import { HashRouter, Switch, Route } from 'react-router-dom';
import { routes } from './routes';
import { DefaultLayout } from './layouts/DefaultLayout';

const CreateRoutes: React.FunctionComponent = () => {
  return (
    <>
      {routes.map(route => (
        <Route key={route.path} path={route.path} component={route.component} exact={route.exact || false} />
      ))}
    </>
  );
};

//TODO: Do we use HashRouter or BrowserRouter??
const Router: React.FunctionComponent = () => {
  return (
    <HashRouter>
      <Switch>
        <DefaultLayout>
          <CreateRoutes />
        </DefaultLayout>
      </Switch>
    </HashRouter>
  );
};

export { Router };
