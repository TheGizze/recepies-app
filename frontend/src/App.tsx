import React from 'react';
import { Router } from './Router';
import './styles/global.scss';

const App: React.FunctionComponent = () => {
  return <Router />;
};

export default App;
