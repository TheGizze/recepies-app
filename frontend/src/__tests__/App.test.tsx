import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';
import { Router } from '../Router';

jest.mock('../Router', () => {
  return {
    __esModule: true,
    Router: jest.fn(),
  };
});

beforeEach(() => {
  jest.resetAllMocks();
});

describe('Root App', () => {
  it('Should match snapshot', () => {
    const mockRouter = Router as jest.Mock;
    mockRouter.mockReturnValue(<div data-testid="mock-routes" />);
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
  it('Should call routes from Router component', () => {
    const mockRouter = Router as jest.Mock;
    mockRouter.mockReturnValue(<div data-testid="mock-routes" />);
    const { findByTestId } = render(<App />);
    expect(findByTestId('mock-routes')).not.toBeNull();
    expect(mockRouter).toHaveBeenCalledTimes(1);
  });
});
