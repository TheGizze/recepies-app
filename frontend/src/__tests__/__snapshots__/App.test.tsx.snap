// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root App Should match snapshot 1`] = `
<div>
  <div
    data-testid="mock-routes"
  />
</div>
`;
